### PR TITLE
Harden file opening and related parsing paths

### PR DESCRIPTION
**Pull request by CoffeeFlux**
## Summary

- Improve validation and resource handling when opening subtitle and associated files
- Tighten handling of linked resources and attachment extraction
- Add regression coverage for parser edge cases

## Testing

- Full macOS build
- C++, LuaJIT, and Automation test suites

### CODE REVIEW

File: src/project.cpp [hunk 1]
```diff
@@ -232,6 +232,43 @@ void Project::LoadUnloadFiles(ProjectProperties properties) {
...
+		}
+		if (avisynth_video && (!avisynth_audio || video != audio)) {
+			script_list += "\n    ";
+			script_list += video.wstring();
+		}
+
+		bool multiple_scripts = avisynth_audio && avisynth_video && audio != video;
+		wxString message = multiple_scripts
+			? _("The subtitle file references AviSynth scripts. AviSynth scripts can execute arbitrary code with your user permissions. Only load them if you trust the subtitle's author.\n\nScripts:")
+			: _("The subtitle file references an AviSynth script. AviSynth scripts can execute arbitrary code with your user permissions. Only load it if you trust the subtitle's author.\n\nScript:");
```
Comment (arch1t3cht) [inline src/project.cpp]: I would also add to the dialog text that the user can still open the subtitle file normally without loading AVS scripts, similarly to the dialog text for local automation scripts.

File: src/video_provider_yuv4mpeg.cpp [hunk 2]
```diff
@@ -114,9 +117,10 @@ class YUV4MPEGVideoProvider final : public VideoProvider {
 
 	int w = 0, h = 0;	/// frame width/height
 	int num_frames = -1; /// length of file in frames
-	int frame_sz;	/// size of each frame in bytes
-	int luma_sz;	/// size of the luma plane of each frame, in bytes
-	int chroma_sz;	/// size of one of the two chroma planes of each frame, in bytes
+	size_t frame_sz = 0;	/// size of each frame in bytes
+	size_t luma_sz = 0;	/// size of the luma plane of each frame, in bytes
+	size_t chroma_sz = 0;	/// size of one of the two chroma planes of each frame, in bytes
+	size_t decoded_frame_sz = 0;	/// size of a decoded BGRA frame in bytes
```
Comment (arch1t3cht) [inline src/video_provider_yuv4mpeg.cpp]: More of a nit, but I've started to stop using unsigned integers for numerical values in accordance with modern wisdom (see e.g. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#res-nonnegative , https://google.github.io/styleguide/cppguide.html#Integer_Types , https://github.com/ericniebler/stl2/issues/182#issuecomment-287683189), which would also mean using `ptrdiff_t` or `int64_t` over `size_t` (and using `std::ssize(foo)` over `foo.size()` to avoid warnings about mixing signed/unsigned)

File: src/ass_parser.cpp [hunk 3]
```diff
@@ -187,13 +193,17 @@ void AssParser::ParseStyleLine(std::string const& data) {
 }
 
 void AssParser::ParseFontLine(std::string const& data) {
-	if (data.starts_with("fontname: "))
+	if (data.starts_with("fontname: ")) {
 		attach = std::make_unique<AssAttachment>(data, AssEntryGroup::FONT);
+		attachment_data = data + "\r\n";
```
Comment (arch1t3cht) [inline src/ass_parser.cpp]: If the `AssAttachment`'s data is collected in `ass_parser` and later set via `SetEntryData` then I think it no longer makes sense to also pass data in the constructor.

I'd either make `AssAttachment`'s constructor no longer take a data argument or delay the `AssAttachment` construction until `FinishAttachment` (removing `attach` and `SetEntryData` and instead also save the attachment's group).

File: src/ass_attachment.cpp [hunk 4]
```diff
@@ -64,7 +64,9 @@ void AssAttachment::Extract(agi::fs::path const& filename) const {
 	auto const& data = entry_data.get();
 	auto header_end = data.find('\n');
 	auto decoded = agi::ass::UUDecode(data.c_str() + header_end + 1, &data.back() + 1);
-	agi::io::Save(filename, true).Get().write(&decoded[0], decoded.size());
+	agi::io::Save save(filename, true);
+	if (!decoded.empty())
+		save.Get().write(decoded.data(), decoded.size());
```
Comment (arch1t3cht) [inline src/ass_attachment.cpp]: Nit: I believe that once `.data()` is used over `&decoded[0]` the if check is unnecessary, i.e. this entire commit can just be replacing `&decoded[0]` with `decoded.data()`.

File: src/ass_parser.cpp [hunk 5]
```diff
@@ -134,7 +134,9 @@ void AssParser::ParseAttachmentLine(std::string const& data) {
 	}
 }
 
-void AssParser::ParseScriptInfoLine(std::string const& data) {
+void AssParser::ParseScriptInfoLine(std::string const& rawdata) {
+	std::string data = SanitizeLine(rawdata);
```
Comment (arch1t3cht) [inline src/ass_parser.cpp]: Is there some concrete issue that this is meant to avoid? When adding the sanitization for the project garbage and extradata sections I intentionally left it out for the remaining sections since null bytes in those *are* technically legal (meaning that VSFilter handles them without problems).

I don't feel too strongly about this since there probably aren't any files depending on this in practice, but if there's no concrete reason to forbid null lines in the script info section I'd tend more towards allowing them.
